nomad/mock/mock.go

```diff
@@ -166,7 +167,7 @@ func Job() *structs.Job {
 		Constraints: []*structs.Constraint{
 			{
 				LTarget: "${attr.kernel.name}",
-				RTarget: "linux",
+				RTarget: runtime.GOOS,
```

Comment:
Will this break local macOS testing where (if I understand correctly) the test runner is macOS but Docker runs Linux images? (I may have that wrong)

I feel like we should either leave this be or switch this job to using the mock driver which *is* cross platform. I feel like it should be mock driver regardless and is only exec for legacy reasons.

Reply:
@notnoop I wouldn't change this. The client doesn't validate this so for testing the client, the mock is still valid even if the OS's do not match, and the scheduler uses the mock job and mock node in conjunction so it will break certain server tests

Reply:
I have replaced this change with a more targetted change in 8a96bf84a1567eee12242dac3c55626b1648ab92 - to fix one case, where a mock job is used but not mock nodes.